contrib/epee/include/net/abstract_http_client.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include <boost/optional/optional.hpp>
+#include "http_base.h"
```

Comment:
I think `http_base.h` and `net_parse_helpers.h` can be moved to the cpp (see other changes). `#incude <string>` is also needed (its being included indirectly somehow now).